### DISCUSSION TITLE
Bug with "content" keyword but can't find why (neovim)

### DISCUSSION THREAD

**Discussion by rambip** [idris-community/idris2-lsp#80] [Q&A]
I just encountered a strange bug in the lsp server.

I use the neovim configuration with built-in lsp, it may be the source of the issue but I'm not sure.

The error is pretty easy to replicate with neovim:

Copy and paste the configuration from the wiki to your vimrc or init.luaf

And then:
```
mkdir error
cd error
echo "package error" > error.ipkg
echo "modules = error" >> error.ipkg

echo "content = error_not_defined" > error.idr 
nvim error.idr
```

I get this strange error:
```
Error executing luv callback:
/usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:601: /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:1
32: invalid header line "content\27[0;1m.\27[0m\
\
\27[38;5;12merror:1:1--1:28\27[0m\
 \27[38;5;12m1 |\27[0m content = error_not_defined\
     \27[38;5;9;1m^^^^^^^^^^^^^^^^^^^^^^^^^^^\27[0m\
\
Content-Length: 317"
```

It turns out the "content" keyword make the lsp communication break at one point, because it appears in the rpc communication.

I don't know how to check what part is wrong, what do you think is more probable:
- This is entirely a `neovim` bug
- This is entirely a `idris2-lsp` bug
- They are both not 100% robust and we're not lucky.

If someone has experience with nvim-lsp, it would be great if he could look what is wrong (or help me debug it).

**Comment by glomurno** [comment]
I've replicated the issue and looked at the logging. If my understanding is correct, is an issue with both client and server.
From the `idris2-lsp` side the issue is upstream with the compiler. Even though the compiler API exposes a functions that builds file and collects the errors, it still print the errors on the stdout while collecting, a behaviour that should be changed.
Normally, the `neovim` client implementation is robust enought to ignore those stray messages on the stdout, however it seams that the `content` or something else in that error messages trips the client.
As a hack this can be potentially solved by redirecting stdout to `/dev/null`, using another channel to output the server messages and then redirect that channel to stdout.

**Comment by rambip** [comment]
So I link to my issue for neovim, I hope that can help.

https://github.com/neovim/neovim/issues/15444
